lib/pysquared/logger.py

```diff
@@ -6,10 +6,71 @@
 import json
 import time
 
+from micropython import const
+
+from lib.pysquared.nvm.counter import Counter
+
+try:
+    from stubs.circuitpython.byte_array import ByteArray
+except ImportError:
+    pass
+
+# NVM register number
+_ERRORCNT = const(7)
+
 
 class Logger:
-    def __init__(self) -> None:
+    def __init__(
+        self,
+        datastore: ByteArray,
+        log_level: str = "DEBUG",
+        log_mode: str = "PRINT",
+    ) -> None:
+        # mapping each level to a numerical value. Used to help support log_level.
+        # If log function used is equal to or above the value, it can be used
+        self.datastore: ByteArray = datastore
+        self.levels_map: dict = {
+            "NOTSET": 0,
+            "DEBUG": 10,
+            "INFO": 20,
+            "WARNING": 30,
+            "ERROR": 40,
+            "CRITICAL": 50,
+        }
+        self.log_modes_set: set = {"PRINT", "FILE", "BOTH"}
```

Comment:
Can we turn these into an ENUM or custom type instead of a string that we parse?